### PR TITLE
Refactor IAnalysisController so its more testable. Also adjust tests in AnalyzerCommandTests.

### PR DESCRIPTION
**Pull request by bigfluffycookie**

### CODE REVIEW

File: Exercise/Exercise.UnitTests/AnalysisControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+using Exercise.Rules;
+using Microsoft.VisualStudio.Setup.Configuration;
+
+namespace Exercise.UnitTests
+{
+    [TestClass]
+    public class AnalysisControllerTests
+    {
+        [TestMethod]
+        public void AnalyzeAndGetResult_MethodsGetCalledCorrectly()
```
Comment (bigfluffycookie) [inline Exercise/Exercise.UnitTests/AnalysisControllerTests.cs:17]: Although I have now made it testable, I am somehow not entirely sure what to test exactly which is why there is only this test method

File: Exercise/Exercise/IAnalysisController.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@ namespace Exercise
 {
     internal interface IAnalysisController
     {
-       void AnalyzeAndGetResult(string fileToAnalyzePath);
+        void AnalyzeAndGetResult(IFile file);
```
Comment (bigfluffycookie) [inline Exercise/Exercise/IAnalysisController.cs:12]: changed this as discussed in meeting

File: Exercise/Exercise.UnitTests/AnalysisControllerTests.cs (hunk 1 above)
Comment (duncanp-sonar) [reply Exercise/Exercise.UnitTests/AnalysisControllerTests.cs:17]: There isn't a lot more you can test in the method; it doesn't have any conditional code, and the `Analyzer` class is still static so you can't easily test that it is being invoked.

One possible additional check that you could do is that issues results are being output (so you would need to have an active known issue).
I wouldn't check the exact formatting of the message - I would just check that the logger was asked to log e.g. the issue id and line number.

**Review by duncanp-sonar — APPROVED** [review]
Looks good. I would add a check that issue results are being logged, but that's all.

**Pull request merged**